src/components/RoomView.tsx

```diff
@@ -224,38 +224,32 @@ function parseDescription (
   roomData: { [roomId: string]: Room },
   presenceData: { [roomId: string]: number }
 ): string {
-  // eslint-disable-next-line no-useless-escape
-  const complexLinkRegex = /\[\[([^\]]*?)\-\>([^\]]*?)\]\]/g
-  const simpleLinkRegex = /\[\[(.+?)\]\]/g
+  // This code used to get room = roomData[roomId], but now that room data is JIT, it
+  // is normal for that to be absent. The links don't actually use data from roomData, so
+  // that is now skipped.
 
+  // Complex "[[text->target]]" links
+  const complexLinkRegex = /\[\[([^\]]*?)->([^\]]*?)\]\]/g
   description = description.replace(complexLinkRegex, (match, text, roomId) => {
-    const room = roomData[roomId]
     const userCount = presenceData[roomId]
     if (roomId === 'item') {
       return `<a class='room-link' href='#' data-item='${text}'>${text}</a>`
     } else if (roomId === 'showModal') {
       return `<a class='room-link' href='#' data-modal='${roomId}'>${text}</a>`
-    } else if (room) {
-      const userCountString = userCount > 0 ? ` (${userCount})` : ''
-      return `<a class='room-link' href='#' data-room='${roomId}'>${text}${userCountString}</a>`
     } else if (linkActions[roomId]) {
       return `<a class='room-link' href='#' data-action='${roomId}'>${text}</a>`
     } else {
-      // TODO: This warning is now expected, with room data being JIT
-      console.log(
-        `Dev warning: tried to link to room ${roomId}, which doesn't exist`
-      )
+      // With room data being JIT, it's now expected that roomData[roomId] might be empty,
```

Comment:
Technically you can have rooms which don't exist in the roomData json but which are valid on the server, but I think in practice we'd want them to be congruent. The behavior of deferring to the server is correct, but I think that for testing purposes the dev warning is still helpful.